Dockerfile.persistent

```diff
@@ -0,0 +1,58 @@
+# --- Build stage ---
+FROM ubuntu:24.04 AS build-stage
+
+RUN apt-get update && apt-get install -y unzip curl build-essential openssl libssl-dev pkg-config && rm -rf /var/lib/apt/lists/*
+
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+
+WORKDIR /opt/rusk
+ENV RUSK_PROFILE_PATH=/opt/dusk/rusk/
+ENV PATH="$PATH:/root/.cargo/bin"
+
+RUN apt-get update && apt-get install -y clang && rm -rf /var/lib/apt/lists/*
+
+COPY . .
+
+ARG TARGETPLATFORM
+# See also https://github.com/docker/buildx/issues/510
+ENV TARGETPLATFORM=${TARGETPLATFORM:-linux/amd64}
+
+# Generate keys and compile genesis contracts
+RUN make keys
+RUN make wasm
+
+# Features to include in the build
+# E.g., --build-arg CARGO_FEATURES="archive"
+ARG NODE_TYPE="provisioner"
+
+RUN case "$NODE_TYPE" in \
+    "provisioner") cargo build --release -p rusk ;; \
+    "archive") cargo build --release --features archive -p rusk ;; \
+    "prover") cargo build --release --no-default-features --features prover -p rusk ;; \
+    *) echo "Unrecognized node type: $NODE_TYPE. Expected one of 'provisioner', 'archive' and 'prover'"; exit 1 ;; \
+    esac
+
+# --- Run stage ---
+FROM ubuntu:24.04 AS run-stage
+
+RUN apt-get update && apt-get install -y unzip curl net-tools libssl-dev && rm -rf /var/lib/apt/lists/*
+    
+WORKDIR /opt/dusk
+
+ENV RUSK_PROFILE_PATH=/opt/dusk/rusk
+ENV RUSK_RECOVERY_INPUT=/opt/dusk/conf/genesis.toml
+ENV RUST_BACKTRACE=full
+ENV NETWORK=mainnet
```

Comment:
Should be parameterized

Reply:
If the user wants a different network, it can be specified with `-e NETWORK=devnet`.
The user is expected to mount a profile path as a volume: `-v /path/to/rusk/profile:/opt/dusk/rusk`. The same can be done for `RUSK_RECOVERY_INPUT`.
Doing it this way allows for a lot of flexibility, as the user can use the same container for different networks with different states.
For example, one state can be used for mainnet:
```
docker run -it \
  -v /path/to/consensus.keys:/opt/dusk/conf/consensus.keys
  -v /mainnet-profile:/opt/dusk/rusk \
  -e NETWORK=mainnet \
  rusk
```

And another can be used for testnet:
```
docker run -it \
  -v /path/to/consensus.keys:/opt/dusk/conf/consensus.keys
  -v /testnet-profile:/opt/dusk/rusk \
  -e NETWORK=testnet \
  rusk
```

Reply:
File and issue to document dockerfile usage in the readme or docs